bctl/daemon/daemon.go

```
@@ -141,15 +141,21 @@ func startServer(logger *bzlogger.Logger, headers map[string]string, params map[
 	if err != nil {
 		return err
 	}
-	cert, err := bzcert.New(logger, config)
+	cert, err := bzcert.New(config)
 	if err != nil {
 		return err
 	}
 
 	// Validate the bzcert before creating the server and fail fast if the cert
 	// is no longer valid. This may result in prompting the user to login again
 	// if the cert contains expired IdP id tokens
-	cert.VerifyAndExitOnError()
+	err = cert.VerifyFromZliConfig()
```

I would prefer if this function was just called `Verify()`, the daemon bzcert cares that it's verifying from the zli config but that should be opaque to code using the daemon bzcert interface.

---

from the rest of your code this seems important on `New()` and `Refresh()` maybe we don't need to expose this function specifically and instead have a new error type on new called `VerificationError`. I don't have beef with how you implemented it but if the validation is key to the functioning of the bzcert then I think it might be nice to centralize that logic? Feel free to ignore tho

---

Yeah i wanted to use Verify but had to name the method something else since i embedded the `IBZCert` interface in the `IDaemonBZCert` interface and go doesnt allow overlapping method names in interfaces.

> from the rest of your code this seems important on New() and Refresh() maybe we don't need to expose this function specifically and instead have a new error type on new called VerificationError. I don't have beef with how you implemented it but if the validation is key to the functioning of the bzcert then I think it might be nice to centralize that logic?

Yeah this actually makes a lot of sense instead of calling `Verify` separately we can always just validate in `New` and `Refresh` which were using a common private method `populateFromConfig` anyway so this is much cleaner. Refactored it to work this way in [368766c](https://github.com/bastionzero/bzero/pull/109/commits/368766c732a1cf0e651d59c6b54b0bd03332cd23). Dont think there is any need for a new error type as we can just bubble up the wrapped error from the verify call in `New` and `Refresh`.